nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -869,47 +894,173 @@ def get_free_port(self):
+        if self.cl_experiment_type == "task":
+            # override the target transform
+            self.dataset_args["target_transform"] = (
+                transforms.Lambda(lambda y: y % self.num_classes_per_task)
+            )
+
+        # whitelist evaluation metrics
+        self.evaluation_metrics = config.get(
+            "evaluation_metrics", ["eval_all_visited_tasks"]
+        )
+        for metric in self.evaluation_metrics:
+            if not hasattr(self, metric):
+                raise ValueError(f"Metric {metric} not available.")
+
+    def should_stop(self):
+        """
+        Whether or not the experiment should stop. Usually determined by the
+        number of epochs but customizable to any other stopping criteria
+        """
+        return self.current_task >= self.num_tasks
+
+    def run_task(self):
+        """Run outer loop over tasks"""
+        # configure the sampler to load only samples from current task
+        print("Training...")
+        self.train_loader.sampler.set_active_tasks(self.current_task)
+
+        # run epochs, inner loop
+        # TODO: return the results from run_epoch
+        for _ in range(self.epochs):
+            self.run_epoch()
+
+        ret = {}
+        for metric in self.evaluation_metrics:
+            eval_function = getattr(self, metric)
+            temp_ret = eval_function()
+            print(temp_ret)
+            for k, v in temp_ret.items():
+                ret[f"{metric}__{k}"] = v
+
+        # TODO: fix this
+        ret.update(
+            timestep_begin=0,
+            timestep_end=1,
+            learning_rate=self.get_lr()[0],
+            extra_val_results=[],
+        )
+
+        self.current_task += 1
+        return ret
+
+    @classmethod
+    def aggregate_results(cls, results):
+        """Run validation in single GPU"""
+        print("Aggregated results")
+        return results[0]
+
+    @classmethod
+    def create_train_sampler(cls, dataset, config):
+        return cls.create_task_sampler(dataset, config, train=True)
+
+    @classmethod
+    def create_validation_sampler(cls, dataset, config):
+        return cls.create_task_sampler(dataset, config, train=False)
+
+    @classmethod
+    def create_task_sampler(cls, dataset, config, train):
+        # assume dataloaders are already created
+        class_indices = defaultdict(list)
+        for idx, (_, target) in enumerate(dataset):
+            class_indices[target].append(idx)
+
+        # defines how many classes should exist per task
+        num_tasks = config.get("num_tasks", 1)
+        num_classes = config.get("num_classes", None)
+        assert num_classes is not None, "num_classes should be defined"
+        num_classes_per_task = math.floor(num_classes / num_tasks)
+
+        task_indices = defaultdict(list)
+        for i in range(num_tasks):   # 1 to 5
+            for j in range(num_classes_per_task):  # 1 to 200
```

Comment:
What do you mean by comments next to `num_tasks` to be `1 to 5` and `num_classes_per_task` to be `1 to 200`?

Reply:
That shouldn't be there, I will remove it.